docs/widget/gas-on-receive.mdx

```diff
@@ -0,0 +1,193 @@
+
+These amounts are designed to cover multiple transactions on the respective chain types.
+
+## Automatic Activation
+
+Gas on Receive automatically activates when:
+- The destination chain is supported
+- The user's destination address has insufficient gas balance (< 3x current gas price)
+- The destination asset is different from the chain's native gas token
+
+Users can manually toggle the feature on/off via the widget interface.
+
+**Cost Impact**: The gas route uses a small portion of your swap amount (e.g., $0.10-$2.00) which slightly reduces your main swap output.
+
+## User Interface
+
+The feature appears in the widget as:
+- **Toggle switch**: Allows users to enable/disable the feature
+- **Gas amount display**: Shows how much gas will be received (e.g., "Enable gas top up - You'll get $2.00 in ETH")
+- **Transaction status**: During execution, shows "Receiving $2.00 in ETH as gas top-up"
+- **Completion status**: After success, displays "Received $2.00 in ETH as gas top-up"
+
+## Configuration
+
+### Widget Configuration
+
+Gas on Receive requires no configuration - the widget auto-detects when it's needed and shows a toggle switch:
+
+```tsx
+import { Widget } from "@skip-go/widget";
+
+function MyApp() {
+  return (
+    <Widget
+      defaultRoute={{
+        srcChainId: "osmosis-1",
+        destChainId: "42161", // Arbitrum  
+        srcAssetDenom: "uosmo",
+        destAssetDenom: "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1" // WETH
+      }}
+      // Widget automatically:
+      // 1. Detects when user lacks gas on destination chain
+      // 2. Shows "Enable gas top up" toggle for supported chains
+      // 3. User manually enables/disables the feature
+    />
+  );
+}
+```
+
+### Client Library Usage
+
+The client library requires manual setup using `executeMultipleRoutes` (v1.5.0+). Use this when building custom interfaces or need more control than the widget provides:
+
+```typescript
+import { executeMultipleRoutes, route } from "@skip-go/client";
+
+// Create your main route and a smaller gas route
+const mainRoute = await route({
+  amountIn: "1000000", // 1 OSMO
+  sourceAssetChainId: "osmosis-1",
+  sourceAssetDenom: "uosmo",
+  destAssetChainId: "42161", // Arbitrum
+  destAssetDenom: "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1" // WETH
+});
+
+const gasRoute = await route({
+  amountIn: "50000", // ~$2 worth for gas  
+  sourceAssetChainId: "osmosis-1",
+  sourceAssetDenom: "uosmo", 
+  destAssetChainId: "42161",
+  destAssetDenom: "0x0000000000000000000000000000000000000000" // Native ETH
+});
+
+// Execute both routes together
+await executeMultipleRoutes({
+  route: { mainRoute, gasRoute },
+  userAddresses: {
+    mainRoute: [
+      { chainId: "osmosis-1", address: "osmo1..." },
+      { chainId: "42161", address: "0x..." }
+    ],
+    gasRoute: [
+      { chainId: "osmosis-1", address: "osmo1..." },
+      { chainId: "42161", address: "0x..." }
+    ]
+  },
+  // Required signing functions
```

Comment:
Let's add slippageTolerancePercent here, because the amount is low, it's better to have high slippage to reduce failed tx
```suggestion
  slippageTolerancePercent: {
      mainRoute: "1",
      feeRoute: "10",
    },
  // Required signing functions
```